core/src/main/scala/alleycats/ConsK.scala

```diff
@@ -0,0 +1,20 @@
+package alleycats
+
+import cats.SemigroupK
+import export.imports
+import simulacrum.typeclass
+
+@typeclass trait ConsK[F[_]] {
+  def cons[A](hd: A, tl: F[A]): F[A]
+}
+
+object ConsK extends ConsK0 {
+  implicit def pureSemigroupKIsConsK[F[_]](implicit p: Pure[F], s: SemigroupK[F]): ConsK[F] =
+    new ConsK[F] {
+      def cons[A](hd: A, tl: F[A]): F[A] = s.combine(p.pure(hd), tl)
+    }
+}
+
+@imports[ConsK]
+trait ConsK0
```

Comment:
We have a huge scala infrastructure at work, so fighting binary incompatibilities in library versions is usually my compelling reason to not expose something that is kind of an internal detail :)